modules/nf-core/yak/main.nf

```diff
@@ -0,0 +1,48 @@
+process YAK {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "bioconda::yak=0.1"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/yak:0.1--h7132678_2':
+        'biocontainers/yak:0.1--h7132678_2' }"
+
+    input:
+    tuple val(meta), path(reads)
+
+    output:
+    tuple val(meta), path("*.yak"), emit: yak
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    yak \\
+        count \\
+        $args \\
+        -t${task.cpus} \\
+        -o ${prefix}.yak <(zcat $reads)
```

Comment:
```suggestion
    if [[ $reads == *.gz ]]; then
        yak \\
            count \\
            $args \\
            -t${task.cpus} \\
            -o ${prefix}.yak <(gzip -dc $reads)
     else
        yak \\
            count \\
            $args \\
            -t${task.cpus} \\
            -o ${prefix}.yak < $reads
     fi            
```

Reply:
To keep the script code shorter I usually do something like:
```nextflow
script:
def useGzip = !( assembly instanceof List ? assembly.every{ it.name.endsWith(".gz") } : assembly.name.endsWith(".gz") )
"""
${useGzip ? 'gzip -c' : 'cat'} $assembly >output.gz
"""
```
The idea is a test is written in groovy, and then the value of it is used in an expression inside the script code to determine which command to use.